internal/configuration/configuration.go

```diff
@@ -107,11 +107,12 @@ func (m *Manager) GetSecrets() models.SecretList {
 }
 
 func (m *Manager) Update(message models.DeviceConfigurationMessage) error {
-	m.lock.RLock()
+	m.lock.Lock()
+	defer m.lock.Unlock()
```

Comment:
I highly disagree on this one TBH, because in this way if the config is finished, the config is locked on all observer, and it'll block GetDeviceID, GetSecrets, GetWorkloads functions if the observers use those ones.

Reply:
Hmmm, you're right; we may run into deadlocks in some cases... I forgot that go locks are not reentrant.